day1/day_1.erl

```diff
@@ -0,0 +1,36 @@
+-module(day_01).
+-export([calc_fuel_for_modules/0, calc_total_fuel/0]).
+
+read_input(FileName) ->
+    {ok, Data} = file:read_file(FileName),
+    lists:delete(<<>>, binary:split(Data, [<<"\n">>], [global])).
```

Comment:
Just things to consider:
1. binary:split with [global] will traverse the whole binary string. For large binaries, that's a lot of time spent
2. lists:delete/2 will similarly traverse the whole list (`O(n)` if you are familiar). 

All to delete the single newline at the end of the binary :) For short inputs this is no big deal but it is important to think through what you're starting with and what the result should be, and to minimize the number of times your code traverses the list/binary as a whole.

I was lazy and did a binary:replace/3 to remove the newline, but if I wasn't lazy today I would just pop off the end of the binary using binary:part/3.

Reply:
Thanks! I just looked into this. I assume if the newline is at the end of the binary I could use part like this to pop it off:

`NewInput = binary:part(Input, {0, byte_size(Input) - 1}).`

Reply:
You got it!